src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs

```diff
@@ -299,20 +299,19 @@ internal static object InternalAlloc(RuntimeType type)
         [LibraryImport(RuntimeHelpers.QCall, EntryPoint = "RuntimeTypeHandle_InternalAlloc")]
         private static unsafe partial void InternalAlloc(MethodTable* pMT, ObjectHandleOnStack result);
 
-        internal static object InternalAllocNoChecks(MethodTable* pMT)
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        internal static object InternalAllocNoChecks(MethodTable* pMT, out uint bytes)
```

Comment:
```suggestion
        [MethodImpl(MethodImplOptions.AggresiveInlining)]
        internal static object InternalAllocNoChecks(MethodTable* pMT)
```
We want to fast path with FCall to be inlined. Instead, the slow path QCall should be in a no-inline method.

Here is an example of a similar fast path / slow path split: https://github.com/dotnet/runtime/blob/5d2444b3e4299dc346dfd8d745fd5057d1dd2554/src/coreclr/System.Private.CoreLib/src/System/RuntimeHandles.cs#L382-L393